app/uk/gov/hmrc/taxenrolmentassignmentfrontend/connectors/EACDConnector.scala

```diff
@@ -131,4 +131,24 @@ class EACDConnector @Inject() (httpClient: HttpClient, logger: EventLoggerServic
         }
       )
   }
+
+  //ES1
+  def getGroupsFromEnrolment(
+    enrolmentKey: String
+  )(implicit hc: HeaderCarrier, ec: ExecutionContext): TEAFResult[GroupsAssignedEnrolment] =
+    EitherT(
+      httpClient.GET[Either[UpstreamErrorResponse, HttpResponse]](
+        s"${appConfig.EACD_BASE_URL}/enrolment-store/enrolments/$enrolmentKey/groups"
+      )
+    )
+      .transform {
+        case Right(response) if response.status == NO_CONTENT => Right(GroupsAssignedEnrolment(List.empty))
+        case Right(response) =>
+          Right(response.json.as[GroupsAssignedEnrolment](GroupsAssignedEnrolment.reads))
```

Comment:
This does match the type of the method: Either[UpstreamErrorResponse, HttpResponse]
Use a HttpResponse in the connector. No need to parse into an object.